File: power-square-android/app/src/main/java/com/okla/ops/beans/ShopPaymentMethod.kt
package com.okla.ops.beans

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class ShopPaymentMethod(
    val address: String? = null,
    val agentName: String? = null,
    val batStock: Int? = null,
    val businessTime: String? = null,
    val cityCode: String? = null,
    val cityName: String? = null,
    val imgList: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val managerPhone: String? = null,
    val name: String? = null,
    val otherPayWay: String? = null,
    val remark: String? = null,
    val saleCashOption: String? = null,
    val saleOnlineOption: String? = null,
    val salePayWay: String? = null,
    val serviceType: String? = null,
    val shopManager: String? = null,
    val shopNo: String? = null,
    val status: Int? = null,
    val type: Int? = null,
    val users: Int? = null,
    val vehicleStock: Int? = null
): Parcelable

File: lib/data/models/shop_payment_method.dart
import 'package:flutter/foundation.dart';

@immutable
class ShopPaymentMethod {
  final String? salePayWay;
  final String? otherPayWay;
  final String? saleCashOption;
  final String? saleOnlineOption;
  final String? shopNo;
  final String? name;

  const ShopPaymentMethod({
    required this.salePayWay,
    required this.otherPayWay,
    required this.saleCashOption,
    required this.saleOnlineOption,
    required this.shopNo,
    required this.name,
  });

  factory ShopPaymentMethod.fromJson(Map<String, dynamic> json) {
    return ShopPaymentMethod(
      salePayWay: json['salePayWay']?.toString(),
      otherPayWay: json['otherPayWay']?.toString(),
      saleCashOption: json['saleCashOption']?.toString(),
      saleOnlineOption: json['saleOnlineOption']?.toString(),
      shopNo: json['shopNo']?.toString(),
      name: json['name']?.toString(),
    );
  }
}
